lnc/lnc.go

```diff
@@ -48,17 +53,33 @@ func NewLNC(store Store) *LNC {
 	}
 }
 
-func (l *LNC) OpenConn(ctx context.Context, passphrase []byte) error {
+func (l *LNC) OpenConn(passphrase string) error {
 	l.mu.Lock()
 	defer l.mu.Unlock()
 
 	if l.conn != nil {
 		return fmt.Errorf("connection already open")
 	}
 
-	session, err := l.store.GetSessionByPassphrase(ctx, passphrase)
-	if err != nil {
+	words := strings.Split(passphrase, " ")
+	if len(words) != mailbox.NumPassphraseWords {
+		return fmt.Errorf("invalid passphrase. Expected %d "+
+			"words, got %d", mailbox.NumPassphraseWords,
+			len(words))
+	}
+
+	var mnemonicWords [mailbox.NumPassphraseWords]string
+	copy(mnemonicWords[:], words)
+	entropy := mailbox.PassphraseMnemonicToEntropy(mnemonicWords)
+
+	ctxb := context.Background()
+	session, err := l.store.GetSessionByPassphrase(ctxb, entropy[:])
```

Comment:
I think we want this to actually be an upsert? So we want the session struct, and it should return or insert depending on if it existed already or not.